jerry-core/vm/vm.c

```diff
@@ -96,7 +96,24 @@ vm_op_get_value (ecma_value_t object, /**< base object */
 
   if (unlikely (ecma_is_value_undefined (object) || ecma_is_value_null (object)))
   {
-    return ecma_raise_type_error (ECMA_ERR_MSG ("Base object cannot be null or undefined."));
+#ifdef JERRY_ENABLE_ERROR_MESSAGES
+    ecma_value_t obj_to_string_result = ecma_op_to_string (object);
+    ecma_string_t *obj_name_p = ecma_get_string_from_value (obj_to_string_result);
```

Comment:
The ecma_op_to_string can have side effects (can run a JS code) and might not be successful. The side effects worries me most, since we might break ES5 compatibility. What about returning with the class of an object? That is always successful.